src/plugins/ui_feature/format.ts

```diff
@@ -28,18 +28,25 @@ export class FormatPlugin extends UIPlugin {
    * first cell (with number type) will be applied to the whole zone.
    */
   private setDecimal(sheetId: UID, zones: Zone[], step: SetDecimalStep) {
-    // Find the first cell with a number value and get the format
-    const numberFormat = this.searchNumberFormat(sheetId, zones);
-    if (numberFormat !== undefined) {
-      // Depending on the step sign, increase or decrease the decimal representation
-      // of the format
-      const newFormat = changeDecimalPlaces(numberFormat, step);
-      // Apply the new format on the whole zone
-      this.dispatch("SET_FORMATTING", {
-        sheetId,
-        target: zones,
-        format: newFormat,
-      });
+    // Find the each cell with a number value and get the format
+    const [zone] = zones;
+    for (let i = zone.top; i <= zone.bottom; i++) {
+      for (let j = zone.left; j <= zone.right; j++) {
+        const numberFormat = this.searchNumberFormat(sheetId, [
```

Comment:
`searchNumberFormat` is now only used for single cell zones. Transform its argument `Zone[]` into `col, row`, and rename it `getCellNumberFormat` or something

Reply:
Changes done